### PR TITLE
Fix issue of navigation to loans to directors additional info

### PR DESCRIPTION
**Pull request by mreis-ch**
Update LoansToDirectorsAdditionalController to implement ConditionalController and add willRender method
Update check in postLoansToDirectorsAdditionalInfo to check if the additional information details field is null and if so create otherwise update

BI-4475

### CODE REVIEW

**Review by ch-tstrickland — COMMENTED** [review]
Is there additional unit testing that could be done?

**Comment by mreis-ch** [comment]
> Is there additional unit testing that could be done?

You're right, yes there could be. Will add some 👍

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/LoansToDirectorsAdditionalInfoControllerTest.java [hunk 1]
```diff
@@ -82,6 +88,8 @@
     private static final String MOCK_CONTROLLER_PATH = UrlBasedViewResolver.REDIRECT_URL_PREFIX + "mockControllerPath";
 
 
+    private static final String COMPANY_ACCOUNTS_DATA_STATE = "companyAccountsDataState";
```
Comment (ch-tstrickland) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/LoansToDirectorsAdditionalInfoControllerTest.java:91]: Please move this constant (and the others while you're at it) to the top, above the other vars.

File: src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/LoansToDirectorsAdditionalInfoControllerTest.java [hunk 2]
```diff
@@ -172,4 +266,26 @@ void postAdditionalInformationBindingResultErrors() throws Exception {
                 .andExpect(status().isOk())
                 .andExpect(view().name(ADDITIONAL_INFORMATION_VIEW));
     }
+
+    @Test
+    @DisplayName("Will render - false")
```
Comment (ch-tstrickland) [inline src/test/java/uk/gov/companieshouse/web/accounts/controller/smallfull/LoansToDirectorsAdditionalInfoControllerTest.java:271]: Could you add more info in this (and the following) DisplayNames? What is the condition that this is testing?

**Review by ch-tstrickland — APPROVED** [review]

**Pull request merged**